File: 2021/12/16/min_operations.js
/**
 * https://leetcode.com/problems/minimum-changes-to-make-alternating-binary-string/
 * @author Andrew Jarombek
 * @since 12/16/2021
 */

const minOperations = function(s) {
    let mismatches = 0;

    for (let i = 0; i < s.length; i++) {
        const c = s.charAt(i);

        if (+s.charAt(i) === i % 2) {
            mismatches++;
        }
    }

    return Math.min(mismatches, s.length - mismatches);
};

File: 2021/12/20/MinOperations.java
/**
 * https://leetcode.com/problems/determine-color-of-a-chessboard-square/
 * @author Andrew Jarombek
 * @since 12/20/2021
 */

public class MinOperations {
    public int minOperations(int[] nums) {
        int result = 0;
        int previous = 0;

        for (int num : nums) {
            int change = Math.max(previous - num + 1, 0);
            result += change;
            previous = num + change;
        }

        return result;
    }
}
